File: assignments/Week03/Day01_(03-02-2020)/Ques2.py
# Question- Write a Python program to combine two dictionary adding values for common keys.
#			d1 = {'a': 100, 'b': 200, 'c':300}
#			d2 = {'a': 300, 'b': 200, 'd':400}
#			Sample output: Counter({'a': 400, 'b': 400, 'd': 400, 'c': 300})

# Answer-
d1 = {'a': 100, 'b': 200, 'c':300}
d2 = {'a': 300, 'b': 200, 'd':400}
print("First Dictionary -> ", d1)
print("Second Dictionary ->", d2)
d2lst = list(d2.keys())
d1lst = list(d1.keys())
for i in d2lst:
	if i not in d1lst:
		newKey = i
d3 = {}
d3.update(d1)
d3[newKey] = 0
for key, value in d3.items():
	for key1 , value1 in d2.items():
		if(key == key1):
			value = value + value1
			d3[key] = value
		else:
			pass
print("Merged Dictionary ->", d3)

File: coding-challenges/Week11/Day03_(01-04-2020)/Ques2.js
/*
Question2- Write a JavaScript program to sort the items of an array.
Sample array :  var arr1 = [ 3, 8, 7, 6, 5, -4, 3, 2, 1 ];
Sample Output : [ -4,3,1,2,3,5,6,7,8]
*/


//Answer-
var array=[-3,8,7,6,5,-4,3,2,1];
var blankArray=[];
var min=array[0];
var pos;
var max=array[0];
for (i=0; i<array.length; i++)
{
        if (max<array[i]) 
        {
        max=array[i];
        }
}

for (var i=0;i<array.length;i++)
{
        for (var j=0;j<array.length;j++)
        {
                if (array[j]!="x")
                {
                        if (min>array[j]) 
                        {
                                min=array[j];
                                pos=j;
                        }
                }
        }
        blankArray[i]=min;
        array[pos]="x";
        min=max;
}
console.log(blankArray);
